src/test/java/com/artipie/rpm/RpmTest.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.rpm;
+
+import com.artipie.asto.Concatenation;
+import com.artipie.asto.Content;
+import com.artipie.asto.Key;
+import com.artipie.asto.Storage;
+import com.artipie.asto.memory.InMemoryStorage;
+import java.nio.charset.Charset;
+import java.nio.file.Path;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.IsEqual;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+/**
+ * Unit tests for {@link Rpm}.
+ *
+ * @since 0.9
+ * @todo #63:30min Don't change metadata when invalid package is sent.
+ *  Currently Rpm is recalculating metadata when an invalid package is sent.
+ *  It should not. Correct that and enable the test below.
+ */
+final class RpmTest {
+
+    /**
+     * Path of repomd.xml fil.
+     */
+    private static final String REPOMD = "repodata/repomd.xml";
+
+    @Test
+    @Disabled
+    void doesntBrakeMetadataWhenInvalidPackageSent(@TempDir final Path tmp)
+        throws Exception {
+        final Storage storage = new InMemoryStorage();
+        final Rpm repo =  new Rpm(
+            storage, StandardNamingPolicy.SHA1, Digest.SHA256, true
+        );
+        storage.save(
+            new Key.From("repo/oldfile.txt"),
+            new Content.From("anything".getBytes())
+        );
+        repo.batchUpdate(Key.ROOT).blockingAwait();
```

Comment:
@paulodamaso `butchUpate` will fail on this line (if you move `repo/oldfile.txt`) to previous line, since it's not rpm package and can not be parsed. So the next lines are always uncreachable.

Reply:
@g4s8 Actually, not. The test code executes as expected, and fails because albeit not adding the invalid package to repository, it updates the revision. PLease take a look:

```
java.lang.AssertionError: 
Expected: "<?xml version=\"1.0\" encoding=\"UTF-8\"?><repomd xmlns=\"http://linux.duke.edu/metadata/repo\"><revision>1588952443</revision><data type=\"primary\"><checksum type=\"sha256\">3cbe6a523d5c5fe7622ce3e3b5f26350d5be0e68098bb2d4d3697dffd3b98e13</checksum><open-checksum type=\"sha256\">c19fe95c089f35d2ed22cf7a9b25c119e122116342667a3d0709405f5b277ba9</open-checksum><location href=\"repodata/924f2a3dc770051cea4b5a09bdc08814c3a5f5a7-primary.xml.gz\"/><size>142</size><open-size>175</open-size></data><data type=\"others\"><checksum type=\"sha256\">6e5ffbf1bd7c27086a9dab793c8ade4040ecdcd032b9a504c12d10c9b55ef3a3</checksum><open-checksum type=\"sha256\">596ae8945eab99b38417cdea5a57fdac12467ccda58ca45824cd1256deb79ca6</open-checksum><location href=\"repodata/c76be7e649e9d9066b93da132f8cc16788c89567-others.xml.gz\"/><size>132</size><open-size>129</open-size></data><data type=\"filelists\"><checksum type=\"sha256\">30e8e3e603de241accc982ffa0b54426dddd378dc62a250689e27a0db5fc3ee8</checksum><open-checksum type=\"sha256\">bac07662a074e658716a47510a027c6e165737851fc116b469ba834a62451189</open-checksum><location href=\"repodata/beb9cbd93f4254a28cd677620b34e2834b7d6840-filelists.xml.gz\"/><size>134</size><open-size>132</open-size></data></repomd>"
     but: was "<?xml version=\"1.0\" encoding=\"UTF-8\"?><repomd xmlns=\"http://linux.duke.edu/metadata/repo\"><revision>1588952444</revision><data type=\"primary\"><checksum type=\"sha256\">3cbe6a523d5c5fe7622ce3e3b5f26350d5be0e68098bb2d4d3697dffd3b98e13</checksum><open-checksum type=\"sha256\">c19fe95c089f35d2ed22cf7a9b25c119e122116342667a3d0709405f5b277ba9</open-checksum><location href=\"repodata/924f2a3dc770051cea4b5a09bdc08814c3a5f5a7-primary.xml.gz\"/><size>142</size><open-size>175</open-size></data><data type=\"others\"><checksum type=\"sha256\">6e5ffbf1bd7c27086a9dab793c8ade4040ecdcd032b9a504c12d10c9b55ef3a3</checksum><open-checksum type=\"sha256\">596ae8945eab99b38417cdea5a57fdac12467ccda58ca45824cd1256deb79ca6</open-checksum><location href=\"repodata/c76be7e649e9d9066b93da132f8cc16788c89567-others.xml.gz\"/><size>132</size><open-size>129</open-size></data><data type=\"filelists\"><checksum type=\"sha256\">30e8e3e603de241accc982ffa0b54426dddd378dc62a250689e27a0db5fc3ee8</checksum><open-checksum type=\"sha256\">bac07662a074e658716a47510a027c6e165737851fc116b469ba834a62451189</open-checksum><location href=\"repodata/beb9cbd93f4254a28cd677620b34e2834b7d6840-filelists.xml.gz\"/><size>134</size><open-size>132</open-size></data></repomd>"
Expected :<?xml version=\"1.0\" encoding=\"UTF-8\"?><repomd xmlns=\"http://linux.duke.edu/metadata/repo\"><revision>1588952443</revision><data type=\"primary\"><checksum type=\"sha256\">3cbe6a523d5c5fe7622ce3e3b5f26350d5be0e68098bb2d4d3697dffd3b98e13</checksum><open-checksum type=\"sha256\">c19fe95c089f35d2ed22cf7a9b25c119e122116342667a3d0709405f5b277ba9</open-checksum><location href=\"repodata/924f2a3dc770051cea4b5a09bdc08814c3a5f5a7-primary.xml.gz\"/><size>142</size><open-size>175</open-size></data><data type=\"others\"><checksum type=\"sha256\">6e5ffbf1bd7c27086a9dab793c8ade4040ecdcd032b9a504c12d10c9b55ef3a3</checksum><open-checksum type=\"sha256\">596ae8945eab99b38417cdea5a57fdac12467ccda58ca45824cd1256deb79ca6</open-checksum><location href=\"repodata/c76be7e649e9d9066b93da132f8cc16788c89567-others.xml.gz\"/><size>132</size><open-size>129</open-size></data><data type=\"filelists\"><checksum type=\"sha256\">30e8e3e603de241accc982ffa0b54426dddd378dc62a250689e27a0db5fc3ee8</checksum><open-checksum type=\"sha256\">bac07662a074e658716a47510a027c6e165737851fc116b469ba834a62451189</open-checksum><location href=\"repodata/beb9cbd93f4254a28cd677620b34e2834b7d6840-filelists.xml.gz\"/><size>134</size><open-size>132</open-size></data></repomd>
Actual   :<?xml version=\"1.0\" encoding=\"UTF-8\"?><repomd xmlns=\"http://linux.duke.edu/metadata/repo\"><revision>1588952444</revision><data type=\"primary\"><checksum type=\"sha256\">3cbe6a523d5c5fe7622ce3e3b5f26350d5be0e68098bb2d4d3697dffd3b98e13</checksum><open-checksum type=\"sha256\">c19fe95c089f35d2ed22cf7a9b25c119e122116342667a3d0709405f5b277ba9</open-checksum><location href=\"repodata/924f2a3dc770051cea4b5a09bdc08814c3a5f5a7-primary.xml.gz\"/><size>142</size><open-size>175</open-size></data><data type=\"others\"><checksum type=\"sha256\">6e5ffbf1bd7c27086a9dab793c8ade4040ecdcd032b9a504c12d10c9b55ef3a3</checksum><open-checksum type=\"sha256\">596ae8945eab99b38417cdea5a57fdac12467ccda58ca45824cd1256deb79ca6</open-checksum><location href=\"repodata/c76be7e649e9d9066b93da132f8cc16788c89567-others.xml.gz\"/><size>132</size><open-size>129</open-size></data><data type=\"filelists\"><checksum type=\"sha256\">30e8e3e603de241accc982ffa0b54426dddd378dc62a250689e27a0db5fc3ee8</checksum><open-checksum type=\"sha256\">bac07662a074e658716a47510a027c6e165737851fc116b469ba834a62451189</open-checksum><location href=\"repodata/beb9cbd93f4254a28cd677620b34e2834b7d6840-filelists.xml.gz\"/><size>134</size><open-size>132</open-size></data></repomd>

```

Reply:
@paulodamaso sorry, I was wrong, `Rpm` ignores `.txt` files and files in `repo/` dir, see previous comment